libs/shared/src/main/java/io/crate/common/collections/CompoundOrdering.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.common.collections;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+
+public final class CompoundOrdering<T> implements Comparator<T> {
```

Comment:
I also checked if we could remove this altogether by switching to chaining `.thenComparing`, but looks like this is a little bit faster:

```
CompoundOrderingBenchmark.benchmarkChainedThenCompare  avgt    6  0.062 ±  0.002  ms/op
CompoundOrderingBenchmark.benchmarkCompoundOrdering    avgt    6  0.057 ±  0.001  ms/op
```

Although close to a measurement error. So not sure if it is worth keeping this around.